packages/cli/src/handlers/dbt/validate.ts

```diff
@@ -0,0 +1,117 @@
+
+const requestValidation = async (projectUuid: string) =>
+    lightdashApi<ApiJobScheduledResponse['results']>({
+        method: 'POST',
+        url: `/api/v1/projects/${projectUuid}/validate?context=cli`,
+        body: undefined,
+    });
+
+const getJobState = async (jobUuid: string) =>
+    lightdashApi<ApiJobStatusResponse['results']>({
+        method: 'GET',
+        url: `/api/v1/schedulers/job/${jobUuid}/status`,
+        body: undefined,
+    });
+
+const getValidation = async (projectUuid: string) =>
+    lightdashApi<ApiValidateResponse['results']>({
+        method: 'GET',
+        url: `/api/v1/projects/${projectUuid}/validate`,
+        body: undefined,
+    });
+
+function delay(ms: number) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+}
+
+const REFETCH_JOB_INTERVAL = 3000;
+
+type ValidateHandlerOptions = {
+    project?: string;
+    verbose: boolean;
+};
+
+const waitUntilFinished = async (jobUuid: string): Promise<string> => {
+    const job = await getJobState(jobUuid);
+    if (job.status === SchedulerJobStatus.COMPLETED) {
+        return job.status;
+    }
+    if (job.status === SchedulerJobStatus.ERROR) {
+        throw new Error(`Validation failed`);
+    }
+
+    return delay(REFETCH_JOB_INTERVAL).then(() => waitUntilFinished(jobUuid));
+};
+
+export const validateHandler = async (options: ValidateHandlerOptions) => {
+    GlobalState.setVerbose(options.verbose);
+    await checkLightdashVersion();
+
+    const config = await getConfig();
+
+    const projectUuid = options.project || config.context?.previewProject;
+
+    if (projectUuid === undefined) {
+        throw new ParameterError(
+            `No project specified, select a project to validate using ${styles.bold(
+                `--project <projectUuid>`,
+            )} or create a preview environment using ${styles.bold(
+                `lightdash star-preview`,
+            )}`,
+        );
+    }
+
+    if (options.project) {
+        console.error(`Validating project ${projectUuid}\n`);
+    } else {
+        console.error(
+            `Validating preview project ${styles.bold(
+                config.context?.previewName,
+            )} with projectUuid ${config.context?.previewProject}\n`,
+        );
+    }
+
+    const validationJob = await requestValidation(projectUuid);
+    const { jobId } = validationJob;
+
+    const spinner = GlobalState.startSpinner(
+        `  Waiting for validation to finish`,
+    );
+
+    await waitUntilFinished(jobId);
+
+    const validation = await getValidation(projectUuid);
+
+    if (validation.length === 0) {
+        spinner?.succeed(`  Validation finished without errors`);
+    } else {
+        spinner?.fail(`  Validation finished with errors\n`);
+
+        validation.forEach((v) => {
+            console.error(
+                `- ${styles.error(v.name)}: ${styles.warning(v.error)}`,
+            );
+        });
```

Comment:
We could use a library like this one to create a table/columns https://www.npmjs.com/package/columnify